### PR TITLE
Escape special characters in the directory paths when calling find_files()

### PR DESCRIPTION
**Pull request by Xiao-Ming**
#### What does this implement/fix? Explain your changes.
In `glob.glob(globstr)`, at `librosa/util/files.py`, line 263, `globstr` contains the given directory name. This can cause errors when calling `librosa.util.find_files(directory)` with `directory` (or its subdirectories) containing special characters "*?[".

This PR fixes it by escaping the special characters in `dir_name` before calling `glob.glob()`.

#### Any other comments?

Error example 1:
image
`librosa.util.find_files("Musics")` will fail because "[2018-06]" is an invalid regular expression.

Error example 2:
image
`librosa.util.find_files("Musics")` will ignore the audio file because the regular expression "[201816]" does not match the folder name.

### CODE REVIEW

**Comment by lostanlen** [comment]
Thanks for this @Xiao-Ming — is there a way you could write a test which triggers this problem on the current version of librosa? maybe by adding a folder named `[test-1511]` or something like that in https://github.com/librosa/librosa-test-data

and making sure that your PR finds the audio file in question

@bmcfee what do you think?

**Comment by bmcfee** [comment]
Thanks @Xiao-Ming , but I think the current behavior is more useful.  If we escape the search pattern, then things like `find_files('*.mp3')` will cease to work.

I'd be open to adding an example in the docstring showing the user how to escape the search pattern prior to calling `find_files`, but I don't think the behavior should change.

**Comment by bmcfee** [comment]
> Indeed users can escape the special characters in the `directory` parameter by themselves, but that does not solve the problem when `recurse=True` and the subdirectories of `directory` contain search patterns.

I don't think this is our problem to solve.  Ultimately, this is a helper utility function, and the more "smarts" we try to inject here, the more likely it is that we run into strange behavior.  (Really, people shouldn't put search characters in filenames, but I digress.)



> Anyway, I agree that some examples regarding search patterns should be added to the docstring. Should I create another PR to address this?

Sure, or you could roll back the commit in this PR and start fresh, that way we keep the discussion in one place.

**Comment by Xiao-Ming** [comment]
Added an example regarding search patterns.

File: librosa/util/files.py [hunk 1]
```diff
@@ -184,6 +184,11 @@ def find_files(
     >>> # Or last 10 files
     >>> files = librosa.util.find_files('~/Music', offset=-10)
 
+    >>> # Avoid including search patterns in the path string
+    >>> directory = '~/[202206] Music'
+    >>> directory = glob.escape(directory)  # Escape the special characters
+    >>> files = librosa.util.find_files(directory)
```
[the remark this answers is not in the document]
Comment (Xiao-Ming) [reply librosa/util/files.py:191]: I think it should be `import glob`. I've added the import line.

**Review by bmcfee — APPROVED** [review]

**Review by bmcfee — APPROVED** [review]

**Pull request merged**